.github/workflows/reusable-ubsan.yml

```diff
@@ -0,0 +1,71 @@
+name: Reusable Undefined Behavior Sanitizer
+
+on:
+  workflow_call:
+    inputs:
+      config_hash:
+        required: true
+        type: string
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  build-ubsan-reusable:
+    name: 'Undefined Behavior sanitizer'
+    runs-on: ubuntu-24.04
+    timeout-minutes: 60
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - name: Runner image version
+      run: echo "IMAGE_OS_VERSION=${ImageOS}-${ImageVersion}" >> "$GITHUB_ENV"
+    - name: Restore config.cache
+      uses: actions/cache@v4
+      with:
+        path: config.cache
+        key: ${{ github.job }}-${{ env.IMAGE_OS_VERSION }}-${{ inputs.config_hash }}
+    - name: Install dependencies
+      run: |
+        sudo ./.github/workflows/posix-deps-apt.sh
+        # Install clang-20
+        wget https://apt.llvm.org/llvm.sh
+        chmod +x llvm.sh
+        sudo ./llvm.sh 20
+    - name: UBSAN option setup
+      run: |
+        echo "UBSAN_OPTIONS=log_path=${GITHUB_WORKSPACE}/ubsan_log" >> "$GITHUB_ENV"
```

Comment:
Just in case it's useful to know, you can do e.g.
```
UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1
```
which means you don't have to pass `-fno-sanitize-recover=*`.

Reply:
Thanks!